### ISSUE TITLE
Client generation request

### ISSUE THREAD

**Issue by sdalezman** [99designs/gqlgen#950]
### What happened?

This is less of an issue and more of an ask to see if there's interest. We have a bunch of graphql API's in production all using gqlgen. 

A lot of those services make requests to each other over gqlgen. It's been ok having to write graphql queries to wire it up. 

One of the frustrating parts is constantly defining some type of struct to use in each service. 

At the same time - we've been experiment with [ent](https://entgo.io/) from facebook. The cool part about ent is the orm models feels pretty close to a lot of the things we're doing to in gqlgen. 

It made me realize that gqlgen is generating a template for the server - why couldn't gqlgen also generate a client to interact with said graphql api. Especially if there are hooks to add authorization etc for however you deal with services to service calls. 

We're thinking about developing some things internally because we do believe it would increase productivity, add more type safety and confidence when interacting with the apis. 

Curious if the gqlgen team has explored the above and/or if others have built and any feedback before we start to write anything on top of gqlgen?

**Comment by vektah** [comment]
dupe of https://github.com/99designs/gqlgen/issues/950

**Comment by carwyn** [comment]
I noticed that #5 was superseded by #740 but I'm not sure if the client generation part was implemented as part of that?

Also see: https://github.com/Yamashou/gqlgenc and [Generate GraphQL Client Code With Go ](https://dev.to/yamashou/generate-graphql-client-code-with-go-5gnd)
